state/pruner.go

```diff
@@ -140,46 +148,32 @@ func (p *Pruner) SetApplicationRetainHeight(height int64) error {
 	return nil
 }
 
-func (p *Pruner) checkHeightBound(height int64) bool {
-	if height <= 0 || height < p.bs.Base() || height > p.bs.Height() {
+func (p *Pruner) heightWithinBounds(height int64) bool {
+	if height < p.bs.Base() || height > p.bs.Height() {
 		return false
 	}
 	return true
 }
 
-// SetCompanionRetainHeight sets the application retain height with some basic
-// checks on the requested height.
+// SetCompanionBlockRetainHeight sets the application block retain height with
+// some basic checks on the requested height.
 //
 // If a higher retain height is already set, we cannot accept the requested
 // height because the blocks might have been pruned.
-//
-// If the application has already set a retain height to a higher value we also
-// cannot accept the requested height as the blocks might have been pruned.
-func (p *Pruner) SetCompanionRetainHeight(height int64) error {
+func (p *Pruner) SetCompanionBlockRetainHeight(height int64) error {
 	// Ensure that all requests to set retain heights via the pruner are
 	// serialized.
 	p.mtx.Lock()
 	defer p.mtx.Unlock()
 
-	if !p.checkHeightBound(height) {
+	if !p.heightWithinBounds(height) {
 		return ErrInvalidHeightValue
 	}
-	currentCompanionRetainHeight, err := p.stateStore.GetCompanionBlockRetainHeight()
+	curRetainHeight, err := p.stateStore.GetCompanionBlockRetainHeight()
 	if err != nil {
```

Comment:
Again here you now assume that an initial block retain height **has** to be set in config otherwise this will always error. I don't have anything against this particular behaviour but we should make this clear in comments above.

Reply:
See https://github.com/cometbft/cometbft/pull/1271/files#diff-85847e2c654966518d4ab21f147cf243a6c7db53f897c5ed6e494ca18b939528R88